sbi/inference/snre/snre_base.py

```diff
@@ -154,6 +147,12 @@ def __call__(
 
         self._handle_x_o_wrt_amortization(x_o, num_rounds)
 
+        # If we're retraining from scratch each round,
+        # keep a copy of the original untrained model for reinitialization.
+        if self._retrain_from_scratch_each_round:
+            self._untrained_posterior = deepcopy(self._posterior)
+            self._untrained_summary_net = deepcopy(self._summary_net)
```

Comment:
We don't save the summary net in other SN*E methods? Sorry if I am overlooking something, this is just judging from diffs here.

Reply:
in SNLE, there is no summary net (the data x are the outputs of the density estimator, so there is no way to encode them). For SNPE, the `summary_net` is part of the `posterior.net` (it is called `embedding_net` there).

Reply:
Insofar as the name difference is not warranted, we should choose one and go with it. I think summary_net plays better with terms that people know from the ABC research, but do you think that it's trainable aspect will confuse people and hence we'd be better advised to use a new term?

Reply:
I would use `embedding_net` to make it more clear that they are **not** summary statistics as in ABC. But yes, let's streamline between SNRE and SNPE